phpBB/includes/functions_content.php

```diff
@@ -327,117 +327,82 @@ function bump_topic_allowed($forum_id, $topic_bumped, $last_post_time, $topic_po
 */
 function get_context($text, $words, $length = 400)
 {
-	// first replace all whitespaces with single spaces
-	$text = preg_replace('/ +/', ' ', strtr($text, "\t\n\r\x0C ", '     '));
+	$output = '';
+	$text_length = utf8_strlen($text);
+
+	// Replace all spaces/invisible characters with single spaces
+	$text = preg_replace("/\s+/", ' ', $text);
 
 	// we need to turn the entities back into their original form, to not cut the message in between them
-	$entities = array('&lt;', '&gt;', '&#91;', '&#93;', '&#46;', '&#58;', '&#058;');
```

Comment:
Is there a reason why you're changing this to be broader and cover all entities? I think there was a point to only decode the characters that need to be decoded.

Reply:
Are you sure?

Reply:
That was written 10 years ago, this is the commit
https://github.com/phpbb/phpbb/commit/433f03107de1162744a1d0b8c9fa3c4ba1b3ed21

I think htmlentities and html_entity_decode do exactly that, and is more complete

Reply:
Maybe I should use the any of the flags and utf-8, but i'm not sure

Reply:
Ah ok, then there was no special point to it. Then this should be fine as it's just used for cutting at a specific length.